tests/kafkatest/sanity_checks/test_mirror_maker.py

```diff
@@ -0,0 +1,98 @@
+        self.zk2 = ZookeeperService(test_context, num_nodes=1)
+
+        self.k1 = KafkaService(test_context, num_nodes=1, zk=self.zk1,
+                                  topics={self.topic: {"partitions": 1, "replication-factor": 1}})
+        self.k2 = KafkaService(test_context, num_nodes=1, zk=self.zk2,
+                                  topics={self.topic: {"partitions": 1, "replication-factor": 1}})
+
+        self.num_messages = 1000
+        # This will produce to source kafka cluster
+        self.producer = VerifiableProducer(test_context, num_nodes=1, kafka=self.k1, topic=self.topic,
+                                           max_messages=self.num_messages, throughput=1000)
+        self.mirror_maker = MirrorMaker(test_context, sources=[self.k1], target=self.k2, whitelist=self.topic)
+
+        # This will consume from target kafka cluster
+        self.consumer = ConsoleConsumer(test_context, num_nodes=1, kafka=self.k2, topic=self.topic,
+                                        consumer_timeout_ms=10000)
+
+    def setUp(self):
+        # Source cluster
+        self.zk1.start()
+        self.k1.start()
+
+        # Target cluster
+        self.zk2.start()
+        self.k2.start()
+
+    def test_lifecycle(self):
+        """Start and stop a single-node MirrorMaker and validate that the process appears and disappears in a
+        reasonable amount of time.
+        """
+        self.mirror_maker.start()
+        node = self.mirror_maker.nodes[0]
+        wait_until(lambda: self.mirror_maker.alive(node), timeout_sec=10, backoff_sec=.5,
+                   err_msg="Mirror maker took too long to start.")
+
+        self.mirror_maker.stop()
+        wait_until(lambda: not self.mirror_maker.alive(node), timeout_sec=10, backoff_sec=.5,
+                   err_msg="Mirror maker took to long to stop.")
+
+    def test_end_to_end(self):
+        """
+        Test end-to-end behavior under non-failure conditions.
+
+        Setup: two single node Kafka clusters, each connected to its own single node zookeeper cluster.
+        One is source, and the other is target.
+
+        - Start mirror maker.
+        - Produce a small number of messages to the source cluster.
+        - Consume messages from target.
+        - Confirm that number of consumed messages matches the number produced.
+        """
+        self.mirror_maker.start()
+        node = self.mirror_maker.nodes[0]
+        wait_until(lambda: self.mirror_maker.alive(node), timeout_sec=10, backoff_sec=.5,
+                   err_msg="Mirror maker took too long to start.")
+
+        self.producer.start()
+        self.producer.wait()
+        self.consumer.start()
+        self.consumer.wait()
+        assert len(self.consumer.messages_consumed[1]) == self.num_messages
+
+        self.mirror_maker.stop()
+        wait_until(lambda: not self.mirror_maker.alive(node), timeout_sec=10, backoff_sec=.5,
+                   err_msg="Mirror maker took to long to stop.")
```

Comment:
Both of these are currently failing for me in a local virtualbox cluster. The first looks like a legit failure since the data doesn't seem to match, the second looks like maybe it's failing to create a topic perhaps because the previous test didn't tear down properly? The log for the second one indicates the topic already exists.

```
======================================================================================================================================================================================================================================================================================================================================================================================================================================
SESSION REPORT (ALL TESTS)
session_id: 2015-08-19--004
run time:   1 minute 40.540 seconds
tests run:  2
passed:     0
failed:     2
======================================================================================================================================================================================================================================================================================================================================================================================================================================
test_id:    2015-08-19--004.kafkatest.sanity_checks.test_mirror_maker.TestMirrorMakerService.test_end_to_end
status:     FAIL
run time:   1 minute 2.808 seconds



Traceback (most recent call last):
  File "/Users/ewencp/confluent/ducktape.git/ducktape/tests/runner.py", line 81, in run_all_tests
    result.data = self.run_single_test()
  File "/Users/ewencp/confluent/ducktape.git/ducktape/tests/runner.py", line 130, in run_single_test
    return self.current_test_context.function(self.current_test)
  File "/Users/ewencp/kafka.git/tests/kafkatest/sanity_checks/test_mirror_maker.py", line 94, in test_end_to_end
    assert len(self.consumer.messages_consumed[1]) == self.num_messages
AssertionError

--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
test_id:    2015-08-19--004.kafkatest.sanity_checks.test_mirror_maker.TestMirrorMakerService.test_lifecycle
status:     FAIL
run time:   37.728 seconds



Traceback (most recent call last):
  File "/Users/ewencp/confluent/ducktape.git/ducktape/tests/runner.py", line 78, in run_all_tests
    self.setup_single_test()
  File "/Users/ewencp/confluent/ducktape.git/ducktape/tests/runner.py", line 122, in setup_single_test
    self.current_test.setUp()
  File "/Users/ewencp/kafka.git/tests/kafkatest/sanity_checks/test_mirror_maker.py", line 58, in setUp
    self.k2.start()
  File "/Users/ewencp/kafka.git/tests/kafkatest/services/kafka.py", line 55, in start
    self.create_topic(topic_cfg)
  File "/Users/ewencp/kafka.git/tests/kafkatest/services/kafka.py", line 115, in create_topic
    node.account.ssh(cmd)
  File "/Users/ewencp/confluent/ducktape.git/ducktape/cluster/remoteaccount.py", line 79, in ssh
    return self._ssh_quiet(self.ssh_command(cmd), allow_fail)
  File "/Users/ewencp/confluent/ducktape.git/ducktape/cluster/remoteaccount.py", line 206, in _ssh_quiet
    raise e
CalledProcessError: Command 'ssh vagrant@worker4 -o 'HostName 127.0.0.1' -o 'Port 2202' -o 'UserKnownHostsFile /dev/null' -o 'StrictHostKeyChecking no' -o 'PasswordAuthentication no' -o 'IdentityFile /Users/ewencp/kafka.git/.vagrant/machines/worker4/virtualbox/private_key' -o 'IdentitiesOnly yes' -o 'LogLevel FATAL'  '/opt/kafka/bin/kafka-topics.sh --zookeeper worker2:2181 --create --topic topic --partitions 1 --replication-factor 1'' returned non-zero exit status 1

--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
```